lib/room.rb

```diff
@@ -0,0 +1,53 @@
+
+
+class Room
+  attr_reader :id
+  attr_accessor :unavailable_dates
+
+  def initialize(id)
+    @id = id
+    @unavailable_dates = []
+  end
+
+
+  def self.create(number_rooms)
+    rooms = []
+    number_rooms.times do |room_number|
+      rooms << Room.new(room_number + 1)
+    end
+    return rooms
+  end
+
+  def self.list
+    return ObjectSpace.each_object(self).to_a
+  end
+
+  def is_available?(check_in,check_out)
+    new_res_dates = (check_in...check_out).to_a
+    unavailable_dates.each do |reservation|
+      reservation = (reservation[:check_in]...reservation[:check_out]).to_a
```

Comment:
Couple of things:
1.  This definitely works and lets you tell if one of the existing reservations conflicts with the given checkin checkout dates.
2.  It's a _little_ inefficient.  You have to generate a bunch of arrays and then check to see if that array contains a given date, which requires another loop.  This could be done with some comparisons to the start and end dates instead.  

BTW #2 is a minor criticism